### ISSUE TITLE
DotNetCoreMSBuildSettings should be analog to MSBuildSettings

### ISSUE THREAD

**Issue by patriksvensson** [cake-build/cake#2544]
Things like `Configuration` is missing.

**Comment by devlead** [comment]
Makes sense, Configurations is also available as root property on below on which has also MSBuildSettings settings.

* [DotNetCoreCleanSettings](https://github.com/cake-build/cake/blob/a0f142af2dfbf18eecc240789529da428cb0d82a/src/Cake.Common/Tools/DotNetCore/Clean/DotNetCoreCleanSettings.cs#L23)
* [DotNetCoreBuildSettings](https://github.com/cake-build/cake/blob/a0f142af2dfbf18eecc240789529da428cb0d82a/src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs#L28)
* [DotNetCorePackSettings](https://github.com/cake-build/cake/blob/a0f142af2dfbf18eecc240789529da428cb0d82a/src/Cake.Common/Tools/DotNetCore/Pack/DotNetCorePackSettings.cs#L23)
* [DotNetCorePublishSettings](https://github.com/cake-build/cake/blob/ffcbbc8f126cc79ceaa1aafd246d36fc894ca562/src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublishSettings.cs#L34)


MSBuildSettings also exists on [DotNetCoreRestoreSettings](https://github.com/cake-build/cake/blob/cb90c043d307a494d52d44a060504cc34542166a/src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestoreSettings.cs#L14) but there's no Config option in the .NET Cli command.
That said `dotnet restore proj/sln /p:Configuration=Debug` is fully valid and supported.

So property should be added.

**Comment by daveMueller** [comment]
I'm currently working on it.
